app/docs/dev.md

```diff
@@ -1,29 +1,207 @@
+- `PER_FOR_DIVISION` Warns about risk of ambiguity when using `per` instead `/` for division.
+- `STRICT_SYNTAX_EXPONENTIATION` Warns that `^` cannot be used for exponentiation when `strict_syntax` is set to `true`.
+- `VALID_CANDIDATE_SET` Message that is displayed when a response is found to be a valid set of groups. **Note:** setting this will not affect the Correct/Incorrect message, it will only add further text.
+- `NOT_DIMENSIONLESS` Message displayed when at least one groups is not dimensionless.
+- `MORE_GROUPS_THAN_REFERENCE_SET` Message displayed when the response contains more groups than necessary.
+- `CANDIDATE_GROUPS_NOT_INDEPENDENT` Message displayed when the groups in the response are not independent.
+- `TOO_FEW_INDEPENDENT_GROUPS` Message displayed when the response contains fewer groups than necessary.
+- `UNKNOWN_SYMBOL` Message displayed when the response contains some undefined symbol.
+- `SUM_WITH_INDEPENDENT_TERMS`  Message displayed when the response has too few groups but one (or more) of the groups is a sum with independent terms.
+
+The evaluation function also has a number of tags for messages that cannot be customised that are used for error messages. These tags and their corresponding message can be found in the `internal_feedback_responses` dictionary in `evaluation.py`. The dictionaries whose keys correspond to customisable feedback messages are listen in `feedback_responses_list` in `evaluation.py`.
+
+## `elementary_functions`
+
+When using implicit multiplication function names with mulitple characters are sometimes split and not interpreted properly. Setting `elementary_functions` to true will reserve the function names listed below and prevent them from being split. If a name is said to have one or more alternatives this means that it will accept the alternative names but the reserved name is what will be shown in the preview.
+
+`sin`, `sinc`, `csc` (alternative `cosec`), `cos`, `sec`, `tan`, `cot` (alternative `cotan`), `asin` (alternative `arcsin`), `acsc` (alternatives `arccsc`, `arccosec`), `acos` (alternative `arccos`), `asec` (alternative `arcsec`), `atan` (alternative `arctan`), `acot` (alternatives `arccot`, `arccotan`), `atan2` (alternative `arctan2`), `sinh`, `cosh`, `tanh`, `csch` (alternative `cosech`), `sech`, `asinh` (alternative `arcsinh`), `acosh` (alternative `arccosh`), `atanh` (alternative `arctanh`), `acsch` (alternatives `arccsch`, `arcosech`), `asech` (alternative `arcsech`), `exp` (alternative `Exp`), `E` (equivalent to `exp(1)`, alternative `e`), `log`, `sqrt`, `sign`, `Abs` (alternative `abs`), `Max` (alternative `max`), `Min` (alternative `min`), `arg`, `ceiling` (alternative `ceil`), `floor`
+
+### `quantities`
+
+String that lists all quantities that can be used in the answer and response.
+
+Each quantity should be written in the form `('quantity name','(units)')` and all pairs concatenated into a single string. See tables below for available default units.
```

Comment:
Is it essential to use 'units', or can we use 'dimensions'? The more fundamental approach would be to define the dimensions of a quantity. Example units is a more superficial, but still complete, way to describe the dimensions of a quantity.

Reply:
Changed behaviour so that using units instead of dimensions is no longer allowed. This simplifies the code somewhat, but will mean that some questions that use comparePhysicalQuantities will require a little extra effort to convert.